applications/connectivity_bridge/src/modules/usb_cdc_handler.c

```diff
@@ -0,0 +1,274 @@
+			poll_dtr();
+		}
+
+		err = k_mem_slab_alloc(&cdc_rx_slab, &rx_buf, K_NO_WAIT);
+		if (err) {
+			data_length = uart_fifo_read(
+				dev,
+				overflow_buf,
+				sizeof(overflow_buf));
+			LOG_WRN("CDC_%d RX overflow", dev_idx);
+		} else {
+			data_length = uart_fifo_read(
+				dev,
+				rx_buf,
+				USB_CDC_RX_BLOCK_SIZE);
+
+			if (data_length) {
+				struct cdc_data_event *event = new_cdc_data_event();
+
+				event->dev_idx = dev_idx;
+				event->buf = rx_buf;
+				event->len = data_length;
+				EVENT_SUBMIT(event);
+			} else {
+				k_mem_slab_free(&cdc_rx_slab, &rx_buf);
+			}
+		}
+	}
+}
+
+static void enable_rx_irq(int dev_idx)
+{
+	uart_irq_callback_user_data_set(
+		devices[dev_idx],
+		cdc_uart_interrupt_handler,
+		(void *)dev_idx);
+	uart_irq_rx_enable(devices[dev_idx]);
+}
+
+static void usbd_status(enum usb_dc_status_code cb_status, const u8_t *param)
+{
+	switch (cb_status) {
+	case USB_DC_ERROR:
+		LOG_ERR("USB_DC_ERROR");
+		break;
+	case USB_DC_RESET:
+		LOG_DBG("USB_DC_RESET");
+		break;
+	case USB_DC_CONNECTED:
+		LOG_DBG("USB_DC_CONNECTED");
+		module_set_state(MODULE_STATE_READY);
+		break;
+	case USB_DC_CONFIGURED:
+		LOG_DBG("USB_DC_CONFIGURED");
+		break;
+	case USB_DC_DISCONNECTED:
+		LOG_DBG("USB_DC_DISCONNECTED");
+		module_set_state(MODULE_STATE_STANDBY);
+		break;
+	case USB_DC_SUSPEND:
+		LOG_DBG("USB_DC_SUSPEND");
+		break;
+	case USB_DC_RESUME:
+		LOG_DBG("USB_DC_RESUME");
+		break;
+	default:
+		break;
+	}
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_uart_data_event(eh)) {
+		const struct uart_data_event *event =
+			cast_uart_data_event(eh);
+		int tx_written;
+
+		if (event->dev_idx >= CDC_DEVICE_COUNT) {
+			return false;
+		}
+
+		if (cdc_ready[event->dev_idx] == 0) {
+			return false;
+		}
+
+		tx_written = uart_fifo_fill(
+			devices[event->dev_idx],
+			event->buf,
+			event->len);
+
+		if (tx_written != event->len) {
+			LOG_DBG("UART_%d->CDC_%d overflow",
+				event->dev_idx,
+				event->dev_idx);
+		}
+
+		return false;
+	}
+
+	if (is_cdc_data_event(eh)) {
+		const struct cdc_data_event *event =
+			cast_cdc_data_event(eh);
+
+		/* All subscribers have gotten a chance to copy data at this point */
+		k_mem_slab_free(&cdc_rx_slab, (void **) &event->buf);
+
+		return true;
+	}
+
+	if (is_module_state_event(eh)) {
+		const struct module_state_event *event =
+			cast_module_state_event(eh);
+
+#if CONFIG_BRIDGE_MSC_ENABLE
+		if (check_state(event, MODULE_ID(fs_handler), MODULE_STATE_STANDBY)) {
+			/* Enable USBD once file system is populated */
+#else
+		if (check_state(event, MODULE_ID(main), MODULE_STATE_READY)) {
+			/* Enable USBD right away */
+#endif
+			int err;
+
+			err = usb_enable(usbd_status);
+			if (err) {
+				LOG_ERR("usb_enable: %d", err);
+				return false;
+			}
+			for (int i = 0; i < CDC_DEVICE_COUNT; ++i) {
+				char dev_name[sizeof(CDC_DEVICE_NAME_TEMPLATE) + 1];
+
+				err = snprintf(
+							dev_name,
```

Comment:
Alignment looks a bit off

Reply:
Looked a lot better in vscode. Same for the indentation comment above

Reply:
Sometimes GitHub messes up the indentation on its own, so it could be okay as is